### PR TITLE
use official docker container for federaated servers

### PR DESCRIPTION
**Pull request by patrickjahns**
## Description
Use official docker container in federated tests

## Motivation and Context
Fix CI

## How Has This Been Tested?
🤖 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -337,6 +337,7 @@ pipeline:
     pull: true
     environment:
       - TEST_SERVER_URL=http://server
+      - TEST_SERVER_FED_URL=http://federated"
```
Comment (individual-it) [inline .drone.yml]: that will make `run.sh` to try to set various settings on the fed. server, in test cases where it was not started it fails.
that was the reason to use `echo "export TEST_SERVER_FED_URL=http://federated" > /drone/saved-settings.sh`

File: .drone.yml [hunk 2]
```diff
@@ -10,40 +10,40 @@ clone:
...
-
-  fix-permissions-federation-server:
-    image: owncloudci/php:${PHP_VERSION}
-    pull: true
-    commands:
-      - chown www-data /drone/fed-server -R
-    when:
-      matrix:
-        USE_FEDERATED_SERVER: true
+#  install-fed-server:
```
Comment (individual-it) [inline .drone.yml]: why not deleting those lines?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: .drone.yml (hunk 2 above)
Comment (patrickjahns) [reply .drone.yml]: Was quickly prototyping - will be deleted when review-ready

**Comment by patrickjahns** [comment]
Issue I am facing now is, that the testing repository would either needed to be moved in a subdirectory to pull the latest changes - or we misuse a github release tag ( 'latest' ) - and re-release the app always on that tag

**Comment by patrickjahns** [comment]
@ownclouders rebase

**Comment by ownclouders** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by ownclouders** [comment]
Automated rebase failed! Please rebase your pull request manually via the command line.

Reason:
```
Command: git rebase base/master

Exit Code: 128

Cause:

error: Failed to merge in the changes.


```

**Comment by patrickjahns** [comment]
Parts of this PR are superseded by #32505 - but once the Guzzle part is finished, we can move with docker containers as federated servers

**Comment by patrickjahns** [comment]
Rebased to fetch latest changes from core/master - afaik we do not require the guzzle part with ssl certificate anylonger

**Comment by PVince81** [comment]
@phil-davis please evaluate if we should continue this and reschedule

**Comment by patrickjahns** [comment]
Went over the current ideas and have the following remarks:
- DIND is not necessary to start the owncloud container when we don't need to add/edit/mount files directly in the container

```
federated:
    image: owncloud/server:latest
    pull: true
    detach: true
    environment:
      - OWNCLOUD_APPS_INSTALL=https://github.com/owncloud/testing/releases/download/latest/testing.tar.gz
      - OWNCLOUD_APPS_ENABLE=testing
    when:
      matrix:
        TEST_FEDERATION: true
```

Now here is the caveat - what type of onwcloud remote instance do we want to test with federation?  Following topics need some more thought:
- should the federated server have redis ( file locking ) or not?
- should we have more than one database system on the federated system ? 


Regarding the issue with ssl:
When you continue to use dind - you could mount a custom apache config and also ssl certificates inside the docker containers and start it with ssl

**Comment by individual-it** [comment]
blocked by https://github.com/owncloud/core/issues/34483

**Comment by individual-it** [comment]
seems pretty complicated to me, closing this in favour of the discussion here https://github.com/owncloud/core/issues/34500
We can reopen this PR or reuse the code if needed

**Pull request closed without merging**